exporter/elasticsearchexporter/factory.go

```diff
@@ -108,3 +115,13 @@ func createTracesExporter(ctx context.Context,
 		exporterhelper.WithShutdown(tracesExporter.Shutdown),
 		exporterhelper.WithQueue(cf.QueueSettings))
 }
+
+// set default User-Agent header with BuildInfo if User-Agent is empty
+func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) {
+	if _, found := cf.Headers[userAgentHeaderKey]; !found {
+		if cf.Headers == nil {
+			cf.Headers = make(map[string]string)
+			cf.Headers[userAgentHeaderKey] = fmt.Sprintf("%s/%s (%s/%s)", info.Description, info.Version, runtime.GOOS, runtime.GOARCH)
+		}
+	}
```

Comment:
I think there is a bug here, if you have Headers with one header (not user-agent) this code will not work). Consider to write the code in a more readable way, to see this:

```golang
	if _, found := cf.Headers[userAgentHeaderKey]; found {
		return
	}
	if cf.Headers == nil {
		cf.Headers = make(map[string]string)
	}
	cf.Headers[userAgentHeaderKey] = fmt.Sprintf("%s/%s (%s/%s)", info.Description, info.Version, runtime.GOOS, runtime.GOARCH)
	}
```

Reply:
nice catch , I will polish here and add more test to cover it.